src/pages/reports/performanceReport/PerformanceReportTable.tsx

```diff
@@ -0,0 +1,224 @@
+    criteria: 'string',
+    criteriaScore: 21,
+    performerType: 'Rock Star',
+    evaluatorFirstName: 'string',
+    evaluatorLastName: 'string',
+    evaluatorPosition: 'stridsfsdfng',
+    evaluateeFirstName: 'string',
+    evaluateeLastName: 'string',
+    evaluateePosition: 'string',
+  },
+  {
+    eventTitle: 'bbs456tring',
+    eventStartsAt: '2012-05-27T16:42:45.838Z',
+    criteria: 'string',
+    criteriaScore: 19,
+    performerType: 'Rock Star',
+    evaluatorFirstName: 'string',
+    evaluatorLastName: 'string',
+    evaluatorPosition: 'string',
+    evaluateeFirstName: 'string',
+    evaluateeLastName: 'string',
+    evaluateePosition: 'sdfsdfsdf ',
+  },
+  {
+    eventTitle: 'bbstrhgjghing',
+    eventStartsAt: '2012-05-27T16:42:45.838Z',
+    criteria: 'string',
+    criteriaScore: 17,
+    performerType: 'Rock Star',
+    evaluatorFirstName: 'string',
+    evaluatorLastName: 'string',
+    evaluatorPosition: 'string',
+    evaluateeFirstName: 'string',
+    evaluateeLastName: 'string',
+    evaluateePosition: 'sdfsdfsdf ',
+  },
+  {
+    eventTitle: 'bbsasdtring',
+    eventStartsAt: '2012-05-27T16:42:45.838Z',
+    criteria: 'string',
+    criteriaScore: 67,
+    performerType: 'Rock Star',
+    evaluatorFirstName: 'string',
+    evaluatorLastName: 'string',
+    evaluatorPosition: 'string',
+    evaluateeFirstName: 'string',
+    evaluateeLastName: 'string',
+    evaluateePosition: 'sdfsdfsdf ',
+  },
+]
+
+const PerformanceReportTable = () => {
+  const [order, setOrder] = React.useState<Order>('asc')
+  const [orderBy, setOrderBy] = React.useState<keyof Data>('eventTitle')
+  const [selected, setSelected] = React.useState<readonly string[]>([])
+  const [page, setPage] = React.useState(0)
+  const [rowsPerPage, setRowsPerPage] = React.useState(5)
+
+  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof Data) => {
+    const isAsc = orderBy === property && order === 'asc'
+    setOrder(isAsc ? 'desc' : 'asc')
+    setOrderBy(property)
+  }
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelecteds = rows.map((n) => n.evaluatorLastName)
+      setSelected(newSelecteds)
+      return
+    }
+    setSelected([])
+  }
+
+  const handleClick = (event: React.MouseEvent<unknown>, name: string) => {
+    const selectedIndex = selected.indexOf(name)
+    let newSelected: readonly string[] = []
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, name)
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1))
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1))
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1),
+      )
+    }
+
+    setSelected(newSelected)
```

Comment:
Concat, slice both are source modyfing functions, I think we should avoid them.

Concat = [...array]
Slice = filter

Reply:
this logic gives us MUI